[init] 도메인 초기 세팅

assokk opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 --> 

- Closes #230 

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- Transaction과 Payment의 엔티티 및 관련 Enum을 정의했습니다.

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)## 💬 공유사항 to 리뷰어<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

assokk committed:
[init] Transaction: 엔티티 및 관련 Enum 정의

assokk committed:
[init] Payment: 엔티티 및 관련 Enum 정의

assokk committed:
[feat] SalePost: Enum 값 추가 및 getSeller 메서드 생성

assokk commented on src/main/java/org/example/ootoutfitoftoday/domain/transaction/entity/Transaction.java:
```diff
@@ -0,0 +1,98 @@
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+@Entity
```
반영완료

assokk commented on src/main/java/org/example/ootoutfitoftoday/domain/payment/entity/Payment.java:
```diff
@@ -0,0 +1,152 @@
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+@Entity
```
반영완료

assokk commented on src/main/java/org/example/ootoutfitoftoday/domain/transaction/entity/Transaction.java:
```diff
@@ -0,0 +1,98 @@
+    @OneToOne(mappedBy = "transaction", cascade = CascadeType.ALL, orphanRemoval = true)
+    private Payment payment;  // 결제 정보
+
+    @Builder(access = AccessLevel.PROTECTED)
```
반영완료

assokk commented on src/main/java/org/example/ootoutfitoftoday/domain/payment/entity/Payment.java:
```diff
@@ -0,0 +1,152 @@
+                .status(PaymentStatus.ESCROWED)
+                .tossOrderId(tossOrderId)
+                .easyPayProvider(easyPayProvider)
+                .build();
```
반영완료

assokk commented on src/main/java/org/example/ootoutfitoftoday/domain/payment/entity/Payment.java:
```diff
@@ -0,0 +1,152 @@
+                .method(PaymentMethod.ACCOUNT_TRANSFER)
+                .status(PaymentStatus.ESCROWED)
+                .tossOrderId(tossOrderId)
+                .build();
```
반영완료

assokk commented on src/main/java/org/example/ootoutfitoftoday/domain/transaction/entity/Transaction.java:
```diff
@@ -0,0 +1,98 @@
+
+    public void attachPayment(Payment p) {
+        this.payment = p;
+    }
```
반영완료

AllSungho reviewed (approved):
수고하셨습니다!

zerone1202 reviewed (approved):
미지님, 정말 고생많으셨습니다.
늦은 시간이라 피곤하실텐데 얼른 주무셔요:)

assokk committed:
[refactor] Transaction: attachPayment 메서드명 수정 및 컨벤션에 따른 코드 수정

- attachPayment를 setPayment로 변경하여 역할을 명확히 드러냄
- @Table 어노테이션 추가
- 빌더 접근 제어 수준을 PRIVATE으로 수정

assokk committed:
[refactor] Payment: 연관관계 설정 및 컨벤션에 따른 코드 수정

- createAccountTransfer와 createEasyPay에 attachTo 메서드를 호출하여 Transaction 객체 간의 양방향 연관관계 설정
- @Table 어노테이션 추가

zerone1202 reviewed (approved):
미지샘 추가로 수정한 부분 확인했습니다.
